src/Foundation/Search/ProductSearch.php

```diff
@@ -179,6 +179,14 @@ public function withinChannels(Channel ...$channels): self
         return $this;
     }
 
+    public function notWithinSkus(array $skus): self
```

Comment:
All good, except for the naming; here's why: "Within" means the product is within a container-like thing, like a channel, taxon, etc. SKUs are unique products so a product can't be within a product represented by an SKU.

This operation is rather an _exclusion_, so I suggest renaming it to `excludingSkus()`.
Also, to make sure every item passed is a string, we can use the variable-length argument PHP feature:

```suggestion
    public function excludingSkus(string  ...$skus): self
```